### ISSUE TITLE
OPTIONS message during call, disable audio RTP

### ISSUE THREAD

**Issue by zucher** [baresip/baresip#3167]
Hi,

during a call , if a SIP OPTIONS is received from a server for keep alive, the stream disable the audio RTP sender and receiver due to call deferencing at the end of ua_handle_options.

```
stream: disable audio RTP sender
stream: disable audio RTP receiver
audio: destroyed (started=0)
stream: disable audio RTP sender
stream: disable audio RTP receiver
```

**Comment by alfredh** [comment]
Can you please provide a SIP trace and additional info mentioned here:

https://github.com/baresip/baresip/wiki/Guidelines-for-reporting-Issues

**Comment by zucher** [comment]
Hi @alfredh ,

**Version**: baresip v3.16.0 , from latest baresip and libre sources

**Description of the problem**: With baresip compiled binary,
 Audio is disabled on OPTIONS message reception during an answer call. Probably the reason also I don't receive DTMF inband digit too.

Logs: 
[accounts.txt](https://github.com/user-attachments/files/17447102/accounts.txt)
[config.txt](https://github.com/user-attachments/files/17447103/config.txt)

[baresip.log](https://github.com/user-attachments/files/17447099/baresip.log)

Account and Config file: 


See attached logs, don't think in the current case wireshark is required. 

Regard

zucher

**Comment by alfredh** [comment]
I think the call object for the active call, and the call object for the incoming OPTIONS handling,
are two individual objects and should not affect each other.

If you are behind NAT and using a SIP-proxy with public address, you need to send out some
audio packets in order to receive some. This can be done with using a real audio device
or e.g. ausine module. You can also try to multiplex RTP/RTCP with
adding `rtcp_mux=yes` in accounts.

You can also try to enable `rtp_stats yes` to see if any RTP packets
was sent or received.

**Comment by alfredh** [comment]
Have you tested a local simple setup first, to make sure that a basic setup is working ?

You can also try this option to open the NAT pinhole:

```
#    ;natpinhole={yes, no}  # default: no
```

@cspiel1 Have you noticed any problems with OPTIONS in your setup ?

**Comment by cspiel1** [comment]
No, not so far.

**Comment by zucher** [comment]
Hi @alfredh ,

I just try your several propositions, from a desk softphone to the baresip device thought the SIP Proxy, dtmf are not received before and after OPTIONS, for both direction.
I initialized after a call from baresip to the sofphone, in band dtmf are well received in both direction, before and after OPTIONS received.

So the conclusion is the OPTIONS calls does not affect an existing call.  so we can close the current issue, I suggest to tag with the call ID the strem enable/disable sequence to have a better context following.

Regarding the dtmf issue, I have to go deeped into the call init sequence regarding the port opening at WAN side. One possible way is to deal with the audio in SDP to force RTP/SAVP on incoming call

Thank you for your help

**Comment by zucher** [comment]
Hi @alfredh 

I found the reason of my issue, 
I try the several flags without results, I found the issue, it was around the RTP/SAVP proxy capabilities, if the encryption is not activated at device declaration side, the rtp link is not well opened on the WAN and no media could be initiated. for outgoing call from bearsip, all was working fine and dtmf  and other audio transmitted, but for incoming calls withtout RTP/SAVP activated nothing working.

OPTIONS is not involved at all , missing may be a tag on enable and disable audio RTP to discriminate calls in logs. So we can close the current issue

Thank you for your support

Regards
